### PR TITLE
Fix named virtual table create statement

### PR DESCRIPTION
**Pull request by Micah-Kolide**
Fixes: #7000 
Fixes: #5714

I noticed some of the named virtual tables don’t appear in the osquery instance. This can be seen by querying `sqlite_temp_master`.

Digging a lot, this is because we add `WITHOUT ROWID` to the column definitions when a table contains a column with either an index or additional flag set. This breaks parsing on the named virtual table create statements.

```
osquery> CREATE VIRTUAL TABLE test USING users() WITHOUT ROWID;
Error: near "WITHOUT": syntax error

osquery> CREATE VIRTUAL TABLE test USING users();

osquery> SELECT uid FROM test LIMIT 1;
+-----+
| uid |
+-----+
| 278 |
+-----+
```

This has been true since we added support for [sqlite 3.14.0](https://github.com/osquery/osquery/commit/763f4e94372c768e3c8f9f235ac5ce7370d2a2c6#diff-496b07a6ace3d0702ddd348f9b0b4fc3f13bcf22c159e2601b23668937d254c0). It wasn’t caught because we ignore the return code. This [Commit](https://github.com/osquery/osquery/commit/0cc9bbbaf7b2687032ee7829f424c19b650986ab) adds some error handling and we can see this. It might make sense to panic, or exit on error here.
```
E0914 10:17:00.132483 -167157632 virtual_table.cpp:1116] Error creating named virtual table: users (1)
E0914 10:17:00.132673 -167157632 virtual_table.cpp:1116] Error creating named virtual table: yara (1)
E0914 10:17:00.132719 -167157632 virtual_table.cpp:1116] Error creating named virtual table: ycloud_instance_metadata (1)
```

This PR additionally removes the column definitions from the `CREATE VIRTUAL TABLE` statement. As these additional arguments are not needed, and don’t appear to be adding much.

```
CREATE VIRTUAL TABLE tablename USING modulename(arg1, arg2, ...);

Each module-argument is passed as written (as text) into the constructor method (xCreate) of the virtual table implementation when the virtual table is created and that constructor is responsible for parsing and interpreting the arguments.

xCreate: If present, the fourth and subsequent strings in the argv[] array report the arguments to the module name in the CREATE VIRTUAL TABLE statement.
```

xCreate could access the column definitions that are in `argv[]`, but instead there’s a second call to `columnDefinition` for `sqlite3_declare_vtab()`.

The only side effect I’ve noticed so far with these changes, is that the schema is not in the `sqlite_temp_master`. But it still appears in `.schema`.

```
osquery> SELECT * FROM sqlite_temp_master WHERE name = 'users';
    type = table
    name = users
    tbl_name = users
    rootpage = 0
    sql = CREATE VIRTUAL TABLE users USING users

osquery> .schema users
CREATE TABLE users(`uid` BIGINT, `gid` BIGINT, `uid_signed` BIGINT, `gid_signed` BIGINT, `username` TEXT, `description` TEXT, `directory` TEXT, `shell` TEXT, `uuid` TEXT, `type` TEXT HIDDEN, `is_hidden` INTEGER, `pid_with_namespace` INTEGER HIDDEN, PRIMARY KEY (`uid`, `username`, `uuid`, `pid_with_namespace`)) WITHOUT ROWID;
```

### CODE REVIEW

File: osquery/sql/virtual_table.cpp [hunk 1]
```diff
@@ -1108,12 +1107,14 @@ Status attachTableInternal(const std::string& name,
...
-        "CREATE VIRTUAL TABLE temp." + name + " USING " + name + statement;
+    auto format = "CREATE VIRTUAL TABLE temp." + name + " USING " + name;
 
     rc =
         sqlite3_exec(instance->db(), format.c_str(), nullptr, nullptr, nullptr);
-
+    if (rc != SQLITE_OK) {
+      LOG(ERROR) << "Error creating named virtual table: " << name << " (" << rc
+                 << ")";
+    }
```
Comment (directionless) [inline osquery/sql/virtual_table.cpp:1117]: I love that you found this omission.

**Review by directionless — DISMISSED** [review]
This makes sense to me.

**Review by directionless — APPROVED** [review]

**Review by directionless — DISMISSED** [review]

**Comment by Smjert** [comment]
This seems to be related to: https://github.com/osquery/osquery/issues/5714 and https://github.com/osquery/osquery/issues/7000

**Comment by directionless** [comment]
One of the comments on #7017 is:
> There are some 'hacks' people use by inspecting sqlite_temp_master and having the schemas there at runtime is helpful.

And this breaks that. 

But, my personal view is that since some tables are missing from `sqlite_temp_master` those hacks already don't work, so there is little value in supporting them. And schemas are visible with `.schema` as well as `table_info` and `table_xinfo` pragmas

**Comment by Smjert** [comment]
(Closing and reopening since some CI fixes landed on master)

**Review by directionless — APPROVED** [review]
Conclusion on office hours is that this PR is an improvement over the broken status quo

**Pull request merged**